app/client/components/identity/idapi/user.ts

```diff
@@ -0,0 +1,26 @@
+import { User } from "../models";
+import { APIUseCredentials, identityFetch } from "./fetch";
+
+export const read = async (): Promise<User> => {
+  const url = "/user/me";
+  const data = await identityFetch(url, APIUseCredentials({}));
+  const consents = data.user.consents
+    .filter((consent: any) => consent.consented)
+    .map((consent: any) => consent.id);
+  return {
+    email: data.user.primaryEmailAddress,
+    consents
+  };
+};
+
+const memoizedRead = (): (() => Promise<User>) => {
```

Comment:
Are there any cases when we need to check back with the origin for `user/me` ? For example if the user has logged out  etc ?

Reply:
Not at this time, and the function name should warn users of its nature. The raw read function can also be exposed in future if needed.